lib/paper_trail/model_config.rb

```diff
@@ -169,6 +161,21 @@ def inherited(klass)
       })
     end
 
+    # Allow .create() events not to revert back to base_class at the final stages when
+    # Association#creation_attributes gets called.  An override for this method from Rails:
+    # https://github.com/rails/rails/blob/v5.2.1/activerecord/lib/active_record/associations/association.rb#L198-L210
+    ActiveRecord::Associations::HasManyAssociation.prepend(Module.new {
+      def creation_attributes
+        if reflection.klass.name == "PaperTrail::Version" &&
+            owner.respond_to?(owner.class.inheritance_column)
+          # For STI records PaperTrail uses the real class name instead of the base_class.
+          { item_type: owner.class.name, item_id: owner[reflection.active_record_primary_key] }
+        else
+          super
+        end
+      end
+    })
```

Comment:
I'm really uncomfortable with patching rails. PT does not currently patch rails at all and I don't want to start.

Reply:
Completely agree -- I'm just trying to keep everyone happy here.  Sean's project uses STI in a way in which he wants changes made to `Management` objects to end up with an `item_type` of `Company` instead, so he suggested a workaround where if there is no defined inheritance_column then it would revert back to pre-1108 behaviour.  This patch is only necessary because of this edge case, combined with a hope to have .joins() work on the `versions` association.

I would LOVE to have total consistency in how PaperTrail operates, and zero reason to have to patch Rails.  Perhaps instead of throwing the baby out with the bathwater in 1143 then we can encourage the use of STI as it was originally conceived -- along with a `type` column so that a single object doesn't temporarily masquerade as another.